### PR TITLE
Task/14820 adapt changes to ppa submission

### PR DESCRIPTION
**Pull request by flxschmidt**
## Description
If the app is in hibernation state, the PPA submission shall be aborted.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-14820

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Services/PPAnalyticsSubmitter/PPAnalyticsSubmitter.swift [hunk 1]
```diff
@@ -128,6 +130,14 @@ final class PPAnalyticsSubmitter: PPAnalyticsSubmitting {
 				completion?(.failure(.appResetError))
 				return
 			}
+			
+			// Hibernation check
+			if strongSelf.cwaHibernationProvider.isHibernationState {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/PPAnalyticsSubmitter/PPAnalyticsSubmitter.swift]: Please move the lines to the beginning of the function. We don't need to wait for the app configuration and all the code above.

File: src/xcode/ENA/ENA/Source/Services/PPAnalyticsSubmitter/__tests__/PPAnalyticsSubmitterTests.swift [hunk 2]
```diff
@@ -538,6 +538,57 @@ class PPAnalyticsSubmitterTests: CWATestCase {
...
+		)
+
+		let expectation = self.expectation(description: "completion handler is called with an error")
+		
+		// WHEN
+		var ppasError: PPASError?
+		sut.triggerSubmitData(ppacToken: nil, completion: { result in
+			switch result {
+			case .success:
+				XCTFail("Test should not success")
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/PPAnalyticsSubmitter/__tests__/PPAnalyticsSubmitterTests.swift]:
```suggestion
				XCTFail("Test should not succeed")
```

**Review by naveeddotio — CHANGES_REQUESTED** [review]
See comments

File: src/xcode/ENA/ENA/Source/Services/PPAnalyticsSubmitter/PPAnalyticsSubmitter.swift (hunk 1 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Services/PPAnalyticsSubmitter/PPAnalyticsSubmitter.swift]: done

File: src/xcode/ENA/ENA/Source/Services/PPAnalyticsSubmitter/__tests__/PPAnalyticsSubmitterTests.swift (hunk 2 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Services/PPAnalyticsSubmitter/__tests__/PPAnalyticsSubmitterTests.swift]: changed

**Review by 30mar — APPROVED** [review]

**Review by naveeddotio — APPROVED** [review]
❤️

**Pull request merged**